dcpy/lifecycle/builds/dbt.py

```diff
@@ -0,0 +1,107 @@
+from pathlib import Path
+import typer
+from ruamel.yaml import YAML
+
+from dcpy.utils.logging import logger
+from dcpy.models.lifecycle.builds import InputDataset, Recipe
+from dcpy.lifecycle.builds import plan
+
+DEFAULT_RECIPE = "recipe.yml"
+
+app = typer.Typer(add_completion=False)
+
+yaml = YAML()
+
+
+class RecipeEditor:
+    """
+    Context manager class that allows recipe files to be edited (and validated)
+    in a with clause. Validation occurs at time of reading and writing.
+    """
+
+    def __init__(self, path):
+        self.path = path
+
+    def __enter__(self):
+        plan.recipe_from_yaml(self.path)
+        with self.path.open() as f:
+            self.recipe = yaml.load(f)
+        return self.recipe
+
+    def __exit__(self, *args):
+        Recipe(**self.recipe)
+        with self.path.open("w") as f:
+            yaml.dump(self.recipe, f)
+
+
+def generate_sources_yml(project_dir: Path, recipe_path: Path = Path(DEFAULT_RECIPE)):
+    """Generate _source.yml from recipe.yml"""
+    raise NotImplementedError()
+
+
+def add_to_recipe_sources(dataset, source):
+    """Small helper to append dataset to yml from _sources.yml"""
+    source["tables"].append({"name": dataset})
+    return source
+
+
+def add_dbt_source(
+    dataset: str,
+    project_dir: Path,
+    recipe_path: Path = Path(DEFAULT_RECIPE),
+    add_staging_stubs: bool = False,
+):
+    """
+    Given dataset name, adds a source dataset to a recipe and _sources.yml
+    Optionally creates a stub sql file in staging folder and adds source to _staging_models.yml
+    """
+    with RecipeEditor(project_dir / recipe_path) as recipe:
+        recipe["inputs"]["datasets"].append(
+            InputDataset(name=dataset).model_dump(
+                exclude_defaults=True, exclude_none=True
+            )
+        )
+
+    sources_path = project_dir / "models" / "_sources.yml"
```

Comment:
not sure if this sources path will remain true. Perhaps make this a global variable for visibility?